oap-server/server-storage-plugin/storage-elasticsearch-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/elasticsearch/query/MetadataQueryEsDAO.java

```diff
@@ -105,6 +105,36 @@ public int numOfConjectural(long startTimestamp, long endTimestamp, int nodeType
         return buildServices(response);
     }
 
+    @Override
+    public List<Database> getAllDatabases() throws IOException {
+        SearchSourceBuilder sourceBuilder = SearchSourceBuilder.searchSource();
+
+        BoolQueryBuilder boolQueryBuilder = QueryBuilders.boolQuery();
+        boolQueryBuilder.must().add(QueryBuilders.termQuery(ServiceInventory.NODE_TYPE, 1));
+
+        sourceBuilder.query(boolQueryBuilder);
+        sourceBuilder.size(100);
+
+        SearchResponse response = getClient().search(ServiceInventory.MODEL_NAME, sourceBuilder);
+
+        List<Database> databases = new ArrayList<>();
+        for (SearchHit searchHit : response.getHits()) {
+            Map<String, Object> sourceAsMap = searchHit.getSourceAsMap();
+            Database database = new Database();
+            database.setId(((Number)sourceAsMap.get(ServiceInventory.SEQUENCE)).intValue());
+            database.setName((String)sourceAsMap.get(ServiceInventory.NAME));
+            String propertiesString = (String) sourceAsMap.get(ServiceInstanceInventory.PROPERTIES);
+            if (!Strings.isNullOrEmpty(propertiesString)) {
+                JsonObject properties = GSON.fromJson(propertiesString, JsonObject.class);
+                if (properties.has(DATABASE)) {
+                    database.setType(properties.get(DATABASE).getAsString());
+                }
```

Comment:
If no `DATABASE`, you need to set `unknown`. Because you design `type` in query protocol is required.